src/java.base/aix/native/libjava/ProcessHandleImpl_aix.c

```diff
@@ -36,9 +38,127 @@
 
 void os_initNative(JNIEnv *env, jclass clazz) {}
 
+/*
+ * Returns the children of the requested pid and optionally each parent and
+ * start time. If requested pid is zero return all processes.
+ * Use getprocs64 to accumulate any process following the rule above.
+ * The resulting pids are stored into the array of longs.
+ * The number of pids is returned if they all fit.
+ * If the parentArray is non-null, store the parent pid.
+ * If the array is too short, excess pids are not stored and
+ * the desired length is returned.
+ */
 jint os_getChildren(JNIEnv *env, jlong jpid, jlongArray jarray,
                     jlongArray jparentArray, jlongArray jstimesArray) {
-    return unix_getChildren(env, jpid, jarray, jparentArray, jstimesArray);
+    pid_t pid = (pid_t) jpid;
+    jlong* pids = NULL;
+    jlong* ppids = NULL;
+    jlong* stimes = NULL;
+    jsize parentArraySize = 0;
+    jsize arraySize = 0;
+    jsize stimesSize = 0;
+    jsize count = 0;
+
+    arraySize = (*env)->GetArrayLength(env, jarray);
+    JNU_CHECK_EXCEPTION_RETURN(env, -1);
+    if (jparentArray != NULL) {
+        parentArraySize = (*env)->GetArrayLength(env, jparentArray);
+        JNU_CHECK_EXCEPTION_RETURN(env, -1);
+
+        if (arraySize != parentArraySize) {
+            JNU_ThrowIllegalArgumentException(env, "array sizes not equal");
+            return 0;
+        }
+    }
+    if (jstimesArray != NULL) {
+        stimesSize = (*env)->GetArrayLength(env, jstimesArray);
+        JNU_CHECK_EXCEPTION_RETURN(env, -1);
+
+        if (arraySize != stimesSize) {
+            JNU_ThrowIllegalArgumentException(env, "array sizes not equal");
+            return 0;
+        }
+    }
+
+    const int chunk = 100;
+    struct procentry64 ProcessBuffer[chunk];
+    pid_t idxptr = 0;
+    int i, num = 0;
+
+    do { // Block to break out of on Exception
+        pids = (*env)->GetLongArrayElements(env, jarray, NULL);
+        if (pids == NULL) {
+            break;
+        }
+        if (jparentArray != NULL) {
+            ppids  = (*env)->GetLongArrayElements(env, jparentArray, NULL);
+            if (ppids == NULL) {
+                break;
+            }
+        }
+        if (jstimesArray != NULL) {
+            stimes  = (*env)->GetLongArrayElements(env, jstimesArray, NULL);
+            if (stimes == NULL) {
+                break;
+            }
+        }
+
+        while ((num = getprocs64(ProcessBuffer, sizeof(struct procentry64), NULL,
+                                 sizeof(struct fdsinfo64), &idxptr, chunk)) != -1) {
+            for (i = 0; i < num; i++) {
+                pid_t ppid = 0;
+                jlong startTime = 0L;
+
+                /* skip files that aren't numbers */
+                pid_t childpid = (pid_t) ProcessBuffer[i].pi_pid;
+                if ((int) childpid <= 0) {
```

Comment:
Can this even happen? Negative pids are group pids, I don't think getprocs returns group pids.